qiskit/qpy/interface.py

```diff
@@ -76,6 +78,7 @@ def dump(
     file_obj: BinaryIO,
     metadata_serializer: Optional[Type[JSONEncoder]] = None,
     use_symengine: bool = True,
+    version: int | None = None,
```

Comment:
Do you think it's worth setting this directly as `version: int = 10`, and us just changing that default on minor releases (if appropriate)?  I'm not sure if allowing `None` buys us anything, and using the value directly makes it appear more obviously in things like documentation and `help`.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11644/commits/2b90f671d49e1407065654c7b1d77568cb8617d3

Reply:
I think the edge case I have in mind is if we have to release a new QPY format in a patch release, but we still want the default to be fixed for the minor release.

If we elect that bugs requiring a QPY format change aren't eligible for backport, then I agree with using the `CURRENT_VERSION` flag, otherwise I prefer a separate `DEFAULT_VERSION` attribute, but I don't feel too strongly.

Reply:
I think not allowing QPY format changes in bugfix releases makes the most sense for a few reasons. I hadn't considered decoupling the latest and default for a potential backport scenario that's an interesting idea. But I'm not sure even then we'd want to allow backports to patch releases because to me it increases the complexity of the backport and increases the risk of making the change.

Reply:
Yeah, in general I agree with this, it's just not hitherto been our practice. If we want to change the policy to this for 1.0+ I'm on board.